src/main/java/edu/ucla/library/iiif/fester/verticles/ManifestVerticle.java

```diff
@@ -328,8 +328,10 @@ private void getLockedManifest(final String aID, final boolean aCollDoc, final P
 
                             aPromise.complete(new LockedManifest(manifest, aCollDoc, lock));
                         } else {
+                            final String errorMessage = LOGGER.getMessage(MessageCodes.MFS_146,
+                                    aCollDoc ? "collection" : "work", aID);
                             lockRequest.result().release();
-                            aPromise.fail(handler.cause());
+                            aPromise.fail(new Throwable(errorMessage));
```

Comment:
It would be better to create an exception here that extends an Exception rather than use a generic Throwable. You can also give that exception the handler.cause() exception as its parent exception so that we don't lose that stacktrace.